src/map/utils/diggingutils.cpp

```diff
@@ -0,0 +1,490 @@
+/*
+===========================================================================
+
+  Copyright (c) 2010-2015 Darkstar Dev Teams
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see http://www.gnu.org/licenses/
+
+  This file is part of DarkStar-server source code.
+
+===========================================================================
+*/
+
+#include "../../common/showmsg.h"
+
+#include <string.h> 
+#include <vector>
+
+#include "../lua/luautils.h"
+#include "../lua/lua_baseentity.h"
+#include "../entities/mobentity.h"
+#include "../../common/timer.h"
+
+#include "../packets/char_skills.h"
+#include "../packets/char_update.h"
+#include "../packets/char_sync.h"
+#include "../packets/chocobo_digging.h"
+#include "../packets/inventory_finish.h"
+#include "../packets/message_text.h"
+#include "../packets/message_special.h"
+#include "../packets/message_standard.h"
+#include "../packets/release.h"
+#include "../packets/message_system.h"
+
+#include "charutils.h"
+#include "diggingutils.h"
+#include "itemutils.h"
+#include "../map.h"
+#include "../time_server.h"
+#include "../vana_time.h"
+#include "zoneutils.h"
+
+/************************************************************************
+*																		*
+*	Chocobo Digging Utilities											*
+*																		*
+************************************************************************/
+
+namespace diggingutils
+{
+
+	// Get message offset for  current zone
+	uint16 m_MessageOffset[3][MAX_ZONEID];
```

Comment:
im sure you dont need this for *every* zone.. it's probably better to make this a std::map of tuples (std::map<uint16, std::tuple<uint16, uint16, uint16> > )